bluesky_widgets/examples/kafka_figures.py

```diff
@@ -0,0 +1,68 @@
+"""
+Run like:
+
+python -m bluesky_widgets.examples.kafka_figures
+
+It will save a png file to ~/bluesky_widgets_test/{run_uid} for each run.
+"""
+from pathlib import Path
+from functools import partial
+import os
+
+import msgpack
+import msgpack_numpy as mpn
+from bluesky_kafka import RemoteDispatcher
+
+from bluesky_widgets.utils.streaming import stream_documents_into_runs
+from bluesky_widgets.models.plot_builders import AutoRecentLines
+from bluesky_widgets.headless.figures import HeadlessFigures
+from bluesky_widgets.examples.utils.generate_msgpack_data import get_catalog
+from bluesky_widgets.models.utils import run_is_live_and_not_completed
+
+
+def export_thumbnails_when_complete(run):
+    "Given a BlueskyRun, export thumbnail(s) to a directory when it completes."
+    model = AutoRecentLines(3)
+    model.add_run(run)
+    view = HeadlessFigures(model.figures)
+
+    home = str(Path.home())
```

Comment:
For examples it's politer to write into `/tmp` so that the stuff you generate doesn't pollute the caller's home directory and so that it doesn't risk clobbering any of their existing files. (Even if the risk of an actual name collision is negligible in a given case, when software does this I wonder what _other_ files in my `$HOME` they might decide to write to, and I worry....)

In this example we make a temporary directory with an random unique name under `/tmp` and then print the files' location.

https://github.com/bluesky/bluesky-widgets/blob/7dbe644d5cc2c2fdb7831650fe76d6025aadcbe6/bluesky_widgets/examples/headless_figures.py#L39-L41

Adding a `print` inside `export` might be useful here too, because they would give feedback on when it has run. (Yesterday we had to watch `ls` to see that.)